packages/discount-code-generator/src/cli.js

```diff
@@ -0,0 +1,217 @@
+
+  .option('help', {
+    alias: 'h',
+  })
+  .help('help', 'Show help text.')
+
+  .option('version', {
+    alias: 'v',
+    type: 'boolean',
+  })
+  .version('version', 'Show version number.', version)
+
+  .option('quantity', {
+    alias: 'q',
+    describe: 'Quantity of discount codes to generate.',
+    demandOption: true,
+  })
+  .coerce('quantity', arg => parseInt(arg, 10))
+
+  .option('length', {
+    alias: 'l',
+    default: 11,
+    describe: 'Length of the discount codes to generate.',
+  })
+  .coerce('length', arg => parseInt(arg, 10))
+
+  .option('prefix', {
+    alias: 'p',
+    default: '',
+    describe: 'Prefix for each code. No prefix will be used if omitted.',
+  })
+
+  .option('inputFile', {
+    alias: 'i',
+    describe: 'Path to code options CSV or JSON file.',
+    demandOption: true,
+  })
+  .coerce('inputFile', (arg) => {
+    if (arg.match(/\.json$/i) || arg.match(/\.csv$/i))
+      return fs.createReadStream(String(arg))
+
+    throw new Error('Invalid input file format. Must be CSV or JSON')
+  })
+
+  .option('outputFile', {
+    alias: 'o',
+    default: 'stdout',
+    describe: 'Path to store generated output file.',
+  })
+  .coerce('outputFile', (arg) => {
+    if (arg === 'stdout')
+      return process.stdout
+    if (arg.match(/\.json$/i) || arg.match(/\.csv$/i))
+      return fs.createWriteStream(String(arg))
+
+    throw new Error('Invalid output file format. Must be CSV or JSON')
+  })
+
+  .option('delimiter', {
+    alias: 'd',
+    default: ',',
+    describe: 'Used CSV delimiter for input and/or output file.',
+  })
+
+  .option('multivalueDelimiter', {
+    alias: 'm',
+    default: ';',
+    describe: 'Used CSV delimiter in multivalue fields for input/output file.',
+  })
+
+  .option('logLevel', {
+    default: 'info',
+    describe: 'Logging level: error, warn, info or verbose.',
+  })
+  .coerce('logLevel', (arg) => {
+    npmlog.level = arg
+  })
+  .argv
+
+// Resolve stream input to javascript object
+const resolveInput = (_args) => {
+  const input = _args.inputFile
+  let _attributes
+  return new Promise((resolve, reject) => {
+    if (input.path.match(/\.json$/i))
+      // Convert input stream to JSON file
+      input
+        .on('error', (error) => {
+          reject(error)
+        })
+        .on('data', (data) => {
+          _attributes = data
+        })
+        .on('end', () => {
+          resolve(JSON.parse(_attributes))
+        })
+    else if (input.path.match(/\.csv$/i))
+      // Convert input stream to CSV file
+      input.pipe(csv({
+        separator: _args.delimiter,
+        strict: true,
+      }))
+        .on('error', (error) => {
+          reject(error)
+        })
+        .on('data', (data) => {
+          const arrayDelim = _args.multivalueDelimiter
+          if (data.cartDiscounts)
+          // eslint-disable-next-line no-param-reassign
+            data.cartDiscounts = data.cartDiscounts.split(arrayDelim)
```

Comment:
Ok, whatever the reason is then maybe add a comment about it next to it, so that other people know why.